mrmustard/lab_dev/states/ket.py

```diff
@@ -180,17 +186,18 @@ def auto_shape(
             max_shape: The maximum shape to compute (default from ``settings.AUTOSHAPE_MAX``).
             respect_manual_shape: Whether to respect the non-None values in ``manual_shape``.
         """
-        if self.ansatz.batch_size <= 1:
+        batch_shape = (
+            self.ansatz.batch_shape[:-1] if self.ansatz._lin_sup else self.ansatz.batch_shape
+        )
+        if batch_shape:  # pragma: no cover
+            raise NotImplementedError("Batched auto_shape is not implemented.")
+        if not self.ansatz._lin_sup:
```

Comment:
the corresponding else raises a wrong warning, and we should revisit how this autoshape is calculated in this case (can you leave the TODO?)